internal/service/ec2/vpc_security_group_test.go

```diff
@@ -2696,7 +2696,7 @@ func TestAccEC2SecurityGroup_rulesDropOnError(t *testing.T) {
 			// Add a bad rule to trigger API error
 			{
 				Config:      testAccSecurityGroupConfig_rulesDropOnError_AddBadRule,
-				ExpectError: regexp.MustCompile("InvalidGroup.NotFound"),
+				ExpectError: regexp.MustCompile("InvalidGroupId.Malformed"),
```

Comment:
Was this test failing previously?

Reply:
I just happened to run into the error myself @bschaatsbergen when testing on the `main` branch. I'm seeing:
```
=== RUN   TestAccEC2SecurityGroup_rulesDropOnError
=== PAUSE TestAccEC2SecurityGroup_rulesDropOnError
=== CONT  TestAccEC2SecurityGroup_rulesDropOnError
    vpc_security_group_test.go:2744: Step 2/3, expected an error with pattern, no match on: Error running apply: exit status 1

        Error: error updating Security Group (sg-019b4813e206e4248): error authorizing Security Group (ingress) rules: InvalidGroupId.Malformed: Invalid id: "sg-malformed" (expecting "sg-...")
        	status code: 400, request id: a479dbd0-2a37-479f-ae8c-94833c880d19

          with aws_security_group.test,
          on terraform_plugin_test.tf line 20, in resource "aws_security_group" "test":
          20: resource "aws_security_group" "test" {

--- FAIL: TestAccEC2SecurityGroup_rulesDropOnError (30.69s)
FAIL
FAIL	github.com/hashicorp/terraform-provider-aws/internal/service/ec2	34.088s
```

Reply:
Yes, this has been failing for a while in nightly CI:

```
== RUN   TestAccEC2SecurityGroup_rulesDropOnError
=== PAUSE TestAccEC2SecurityGroup_rulesDropOnError
=== CONT  TestAccEC2SecurityGroup_rulesDropOnError
vpc_security_group_test.go:2683: Step 2/3, expected an error with pattern, no match on: Error running apply: exit status 1
Error: error updating Security Group (sg-015bb0d99616e28a1): error authorizing Security Group (ingress) rules: InvalidGroupId.Malformed: Invalid id: "sg-malformed" (expecting "sg-...")
  status code: 400, request id: 401b8687-294c-484a-a5fb-f5a2ce7fa5d4
with aws_security_group.test,
on terraform_plugin_test.tf line 20, in resource "aws_security_group" "test":
20: resource "aws_security_group" "test" {
--- FAIL: TestAccEC2SecurityGroup_rulesDropOnError (97.27s)
```